src/types/article.ts

```diff
@@ -239,13 +239,42 @@ export default /* GraphQL */ `
     "whether readers can comment"
     canComment: Boolean!
 
+    "history versions"
+    versions(input: ArticleVersionsInput!): ArticleVersionsConnection! @complexity(multipliers: ["input.first"], value: 1)
+
     ##############
     #     OSS    #
     ##############
     oss: ArticleOSS! @auth(mode: "${AUTH_MODE.admin}")
     remark: String @auth(mode: "${AUTH_MODE.admin}")
   }
 
+  input ArticleVersionsInput {
+    after: String
+    first: Int @constraint(min: 0)
+  }
+
+  type ArticleVersionsConnection implements Connection {
+     totalCount: Int!
+     pageInfo: PageInfo!
+     edges: [ArticleVersionEdge]!
+  }
+
+  type ArticleVersionEdge {
+     node: ArticleVersion!
+     cursor: String!
+  }
+
+  type ArticleVersion implements Node {
```

Comment:
need to revise `node` and `nodes` resolvers to support `ArticleVersion`

Reply:
[here](https://github.com/thematters/matters-server/pull/3798/files#diff-de19d85571b95ec048a6d56e10889ee21253a457e249ac35f81589238ed2b903R14) is